dof_processor.py

```diff
@@ -0,0 +1,589 @@
+                    logging.warning(f"LibreOffice process cleanup completed after timeout of {doc_path.name}")
+                
+                return None
+                
+        except Exception as e:
+            logging.error(f"Error during conversion of {doc_path}: {e}")
+            return None
+
+def get_problematic_files() -> List[str]:
+    """
+    Returns the list of files that reached timeout during conversion
+    
+    Returns:
+        List of problematic file paths
+    """
+    return problematic_files.copy()
+
+def clear_problematic_files() -> None:
+    """
+    Clears the list of problematic files
+    """
+    global problematic_files
+    problematic_files.clear()
+
+def save_problematic_files_report(output_dir: Path) -> Optional[Path]:
+    """
+    Saves a report of problematic files to a file
+    
+    Args:
+        output_dir: Directory where to save the report
+        
+    Returns:
+        Path to the created report file or None if no problematic files
+    """
+    global problematic_files
+    
+    if not problematic_files:
+        return None
+    
+    report_path = output_dir / f"archivos_problematicos_{datetime.now().strftime('%Y%m%d_%H%M%S')}.txt"
+    
+    try:
+        with open(report_path, 'w', encoding='utf-8') as f:
+            f.write(f"Problematic Files Report - {datetime.now().strftime('%Y-%m-%d %H:%M:%S')}\n")
+            f.write("=" * 80 + "\n\n")
+            f.write(f"Total files that reached timeout ({CONVERSION_TIMEOUT_SECONDS}s): {len(problematic_files)}\n\n")
+            
+            for i, file_path in enumerate(problematic_files, 1):
+                f.write(f"{i:3d}. {file_path}\n")
+        
+        logging.info(f"Problematic files report saved at: {report_path}")
+        return report_path
+        
+    except Exception as e:
+        logging.error(f"Error saving problematic files report: {e}")
+        return None
+
+
+def merge_docx_files(docx_files: List[Path], output_path: Path) -> bool:
+    """
+    Merges multiple DOCX files into a single document using docxcompose
+    
+    Args:
+        docx_files: List of paths to DOCX files (can be strings or Path objects)
+        output_path: Path to the merged output file (can be string or Path object)
+        
+    Returns:
+        True if merge was successful, False otherwise
+    """
+    if not docx_files:
+        logging.warning("No DOCX files to merge.")
+        return False
+    
+    existing_files = []
+    for f in docx_files:
+        path_obj = Path(f) if isinstance(f, str) else f
+        if path_obj.exists():
+            existing_files.append(path_obj)
+    
+    if not existing_files:
+        logging.warning("No valid DOCX files found to merge.")
+        return False
+    
+    output_path = Path(output_path) if isinstance(output_path, str) else output_path
+    
+    try:
+        logging.info(f"Starting merge of {len(existing_files)} DOCX files using docxcompose...")
```

Comment:
let's try to avoid try within try. I would remove this one, have we ran into this errors? Maybe we should catch them in the caller function `process_date_edition`, because it has a if/else that does the same that could be done in a try/catch ;-)